geode-core/src/main/java/org/apache/geode/internal/cache/DistributedRegion.java

```diff
@@ -1853,8 +1880,10 @@ void distributeInvalidate(EntryEventImpl event) {
   void basicUpdateEntryVersion(EntryEventImpl event) throws EntryNotFoundException {
     InternalRegion internalRegion = event.getRegion();
     AbstractRegionMap regionMap = (AbstractRegionMap) internalRegion.getRegionMap();
+    boolean locked = false;
     try {
       regionMap.lockForCacheModification(internalRegion, event);
+      locked = internalRegion.lockWhenRegionIsInitializing();
```

Comment:
see other comments. I know you are adding calls every place we do lockForCacheModification but I do wonder if this one is needed while initializing. Better safe than sorry, but I was wondering if you know if basicUpdateEntryVersion needed to be prevented during cleanup after a failed GII